test/unit-test/test_derivation.ml

```diff
@@ -28,6 +28,61 @@ let%expect_test "derivation" =
       (Conj (Empty Empty))))
   |}];
 
+  (*
+   * case 42 of
+   *   X when true -> X
+   * end
+   *)
+  print Context.empty (Case (Constant (Number 42), [(PatVar "x", Constant (Atom "true")), Var "x"]));
+  [%expect {|
+     (Ok (
+       (TyVar a)
+       (Conj (
+         (Disj (
+           Empty (
+             Conj (
+               (Eq
+                 (TyVar a)
+                 (TyVar b))
+               (Eq (TySingleton (Number 42)) (TyVar b))
+               Empty
+               Empty
+               Empty))))
+         Empty))))
+  |}];
+
+
+  (*
+   * case {41, 42} of
+   *   {X, Y} when true -> {X, Y}
+   * end
+   *)
+  print Context.empty (Case (Tuple [Constant (Number 41); Constant (Number 42)], [(PatTuple [PatVar "x"; PatVar "y"], Constant (Atom "true")), Tuple [Var "x"; Var "y"]]));
+  [%expect {|
+     (Ok (
+       (TyVar a)
+       (Conj (
+         (Disj (
+           Empty (
+             Conj (
+               (Eq
+                 (TyVar a)
+                 (TyTuple (
+                   (TyVar b)
+                   (TyVar c))))
+               (Eq
+                 (TyTuple (
+                   (TySingleton (Number 41))
+                   (TySingleton (Number 42))))
+                 (TyTuple (
+                   (TyVar b)
+                   (TyVar c))))
+               (Conj (Empty Empty))
+               Empty
+               (Conj (Empty Empty))))))
+         (Conj (Empty Empty))))))
+  |}];
+
```

Comment:
Could you add test cases for multiple clauses?

```
case 42 of
  N when false -> N;
  N when true -> N
end
```

Reply:
OK.  I'll add such test cases.

Reply:
It's fixed in the commit https://github.com/kmizu/fialyzer/commit/f222607d2157d94e02260953a786a8899ab6da56

Reply:
Thank you!